src/pybind/mgr/dashboard/frontend/src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -40,6 +40,24 @@ export class NavigationComponent implements OnInit, OnDestroy {
   displayedSubMenu = {};
   private subs = new Subscription();
 
+  private groupedPaths: Record<string, string[]> = {
+    cluster: [
+      'pool',
+      'hosts',
+      'osd',
+      'inventory',
+      'crush-map',
+      'monitor',
+      'expand-cluster',
+      'perf_counters'
+    ],
+    block: ['block'],
+    object: ['rgw'],
+    file: ['cephfs', 'nfs'],
+    observe: ['logs', 'monitoring'],
+    admin: ['services', 'upgrade', 'ceph-users', 'mgr-modules', 'configuration', 'cluster']
```

Comment:
I'm a bit afraid of duplicating definitions: this might force developers to keep it in sync in 2 different places: here and in the HTML template... (I'd personally go for this JSON object, rather than the template, but then [I'd remove this from the Angular HTML and dynamically generate the sidenav](https://github.com/ceph/ceph/pull/33375) menu from this structure... however, it's not such a trivial task).

Anyway, given this hierarchy is already encoded in the HTML side, as a nested `<ul><li></li></ul>` collection, and that Angular is already iterating over the DOM whenever a user clicks on a link or types a URL (activated route), can't we somehow leverage what Angular is already doing for us?

(NOTE: I'm a fan of minimizing the amount of components TS code in Angular: whenever something can be done in the HTML template, CSS, Pipes, etc., my vote is for that approach).

Reply:
@epuertat 

The reason this this object is what [I explained in my previous comment](https://github.com/ceph/ceph/pull/54043#discussion_r1365322852): the routes grouping don't match the html left side navigation grouping.

Reply:
> I'm a bit afraid of duplicating definitions: this might force developers to keep it in sync in 2 different places: here and in the HTML template... 

Actually, the current architecture of the application forces the users to keep routes in sync in **multiple** places, not one. If you change it in one place you have to place in many other depending on the route. Including the backend for one case.

Reply:
@epuertat 

I have refactored the routes as discussed above and removed the routes map from ```navigation.component.ts```. The one thing that will remain is the ```toggleSubMenu()``` in order to track the state of ```ngbCollapse```. Replacing it by matching URLs to the collapsed state of the side menu items would result in poor UX since that would force users to make a HTTP request + screen render just to see the contents of the sub menu. Other than that the functionality is working as expected. When you refresh the screen the side navigation section will un-collapse to match the base route.